src/Symfony/Component/Validator/Constraints/UrlValidator.php

```diff
@@ -61,7 +61,13 @@ public function validate($value, Constraint $constraint)
             return;
         }
 
-        $pattern = sprintf(static::PATTERN, implode('|', $constraint->protocols));
+        $protocolPattern = '(%s):';
+        if ($constraint->relativeProtocol) {
+            $protocolPattern = '((%s):)?';
```

Comment:
What about `(?:'.$protocolPattern.')?` to reuse previous pattern. Also no real reason for capturing here.